spec/unit/provider/git_spec.rb

```diff
@@ -595,6 +595,12 @@
     expect { @provider.run_action(:sync) }.to raise_error(Chef::Exceptions::MissingParentDirectory)
   end
 
+  it "raises an error if the user does not exist" do
+    @resource.user("whois")
```

Comment:
Right now this test is actually hitting the system to see if it can find the whois user. We've had issues with tests like this in the past when those users are actually created. They result in painful troubleshooting sessions we should try to avoid. You should probably mock out the reponse from get_dir so we can just test the assertion portion here without hitting the actual system.